parlai/agents/hugging_face/gpt2.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.torch_generator_agent import TorchGeneratorAgent, TorchGeneratorModel
+from parlai.agents.hugging_face.dict import Gpt2DictionaryAgent
+from parlai.utils.misc import warn_once
+from parlai.utils.torch import DummyEncoder, concat_without_padding
+
+from transformers import GPT2Model
+
+import torch
+
+############################################
+## Modules
+############################################
+
+
+class GPT2Decoder(torch.nn.Module):
```

Comment:
a very brief 2 sentences on the high level (We use a dummy encoder because ...) *could* be helpful at the top here.